crates/oci/src/client.rs

```diff
@@ -266,6 +266,24 @@ impl Client {
 
             layers.push(layer);
 
+            let mut deps = BTreeMap::default();
+            for (dep_name, mut dep) in c.dependencies {
```

Comment:
Is this always assuming that dependencies are located on disk?

Reply:
Hmm, by this stage everything not on disk _should_ have been resolved to file paths in the cache - that's what happens for primary components coming from the network.  But you're right, dependencies travel their own road, so I need to verify that assumption.  Thanks for the catch!

Reply:
Yeah I think we're good.  It's fiddly to test for sure given that a lurking remote reference would presumably still get resolved without much deleting, but this makes me pretty confident:

```toml
[component.spin-comp-deps-test-two]
source = "spin_comp_deps_test.wasm"
[component.spin-comp-deps-test-two.dependencies]
"vscode:example/types" = { package = "vscode:example", version = "0.0.2", registry = "registrytest-vfztdiyy.fermyon.app" }
```

```json
// lockfile
    {
      "id": "spin-comp-deps-test-two",
      "source": {
        "content_type": "application/wasm",
        "source": "file:///home/ivan/testing/target-env-test/spin_comp_deps_test.wasm"
      },
      "dependencies": {
        "vscode:example/types": {
          "source": {
            "content_type": "application/wasm",
            "source": "file:///home/ivan/.cache/spin/registry/wasm/sha256:fcc77e6020ee40ffcc90667f60605adb3bc050b93e86e1086d1ff307bf7e80ef"
          },
          "export": null
        }
      }
    }
```

```json
// OCI config
    {
      "id": "spin-comp-deps-test-two",
      "source": {
        "content_type": "application/wasm",
        "digest": "sha256:7acaaaab00c5c08e5563656eca8a52e7d718c3da775d298b2b222876f76677ec"
      },
      "dependencies": {
        "vscode:example/types": {
          "source": {
            "content_type": "application/wasm",
            "digest": "sha256:fcc77e6020ee40ffcc90667f60605adb3bc050b93e86e1086d1ff307bf7e80ef"
          },
          "export": null
        }
      }
    }
```